cdap-app-fabric/src/main/java/io/cdap/cdap/master/environment/k8s/AbstractServiceMain.java

```diff
@@ -178,6 +181,12 @@ protected void configure() {
                 new SupplierProviderBridge<>(masterEnv.getDiscoveryServiceClientSupplier()));
       }
     });
+    modules.add(new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(AuditLogWriter.class).to(MessagingAuditLogWriter.class).in(Scopes.SINGLETON);
```

Comment:
Yep, looks like having a module, potentially within RuntimeModule makes sense. Saves from making unnamed modules in place

Reply:
This would also allow more clean disabling by providing noop module if the FF is disabled